### ISSUE TITLE
Scala 2.13 duplicates items on a java set conversion map apply

### ISSUE THREAD

**Issue by csaltos** [scala/bug#11629]
* HOW TO REPRODUCE IT:

Using Scala 2.13.0 run:

```
import scala.jdk.CollectionConverters._
import java.util
val jSet = new util.HashSet[String]()
jSet.add("hello")
jSet.asScala.map(println)
```

* CURRENT BEHAVIOUR:

"hello" is printed twice

* DESIRED BEHAVIOUR:

"hello" is printed only once (since there's only one item on the set, not two)

NOTE: this not only happens with print but also with any function (it's called twice per item mapped)

NOTE2: this also happens with the deprecated scala.collection.JavaConverters and the new scala.jdk.CollectionConverters (both under Scala 2.13.0, of course)

NOTE3: this also happens with more elements, if you have 2 elements on the Set, then 4 calls are done, if you have 3 then 6, etc.

NOTE4: you may also use this sample project ready to go at -> https://github.com/csaltos/jsetpossiblebug

(this is first reported by Jim Balhoff at -> https://users.scala-lang.org/t/possible-bug-in-javaconverters-set-iteration/4826)

**Comment by balhoff** [comment]
Thanks for posting this!

**Comment by Jin-H** [comment]
Perhaps I found the reason why this code println twice.
I debug this code and found after `xs` match `_`, duplicate invoke super.addAll(xs),which result is this.

```scala

  override def addAll(xs: IterableOnce[A]): this.type = {
    sizeHint(xs.knownSize)
    xs match {
      case hm: immutable.HashSet[A] =>
        hm.foreachWithHash((k, h) => addElem(k, improveHash(h)))
        this
      case hm: mutable.HashSet[A] =>
        val iter = hm.nodeIterator
        while (iter.hasNext) {
          val next = iter.next()
          addElem(next.key, next.hash)
        }
        this
      case _ => super.addAll(xs)
    }
   // which result in duplicate println
    super.addAll(xs)
  }
```

**Comment by Jin-H** [comment]
The code I paste was in `scala.collection.mutable.HashSet`.
```scala
override def addAll(xs: IterableOnce[A]): this.type = {
    sizeHint(xs.knownSize)
    xs match {
      case hm: immutable.HashSet[A] =>
        hm.foreachWithHash((k, h) => addElem(k, improveHash(h)))
      case hm: mutable.HashSet[A] =>
        val iter = hm.nodeIterator
        while (iter.hasNext) {
          val next = iter.next()
          addElem(next.key, next.hash)
        }
      case _ => super.addAll(xs)
    }
// I think this code should be replace with `this`
       this
//    super.addAll(xs)
  }
```

**Comment by joshlemer** [comment]
I believe this is a duplicate of https://github.com/scala/bug/issues/11601

**Comment by som-snytt** [comment]
Thanks. Hopefully tooling around duplication of effort will improve.
